commit 2546c60004309ede8e2d1d5341e0decd90e057bf
Author: David E. Box <david.e.box@linux.intel.com>
Date:   Fri Feb 11 17:32:50 2022 -0800

    platform/x86: Add Intel Software Defined Silicon driver
    
    Intel Software Defined Silicon (SDSi) is a post manufacturing mechanism for
    activating additional silicon features. Features are enabled through a
    license activation process.  The SDSi driver provides a per socket, sysfs
    attribute interface for applications to perform 3 main provisioning
    functions:
    
    1. Provision an Authentication Key Certificate (AKC), a key written to
       internal NVRAM that is used to authenticate a capability specific
       activation payload.
    
    2. Provision a Capability Activation Payload (CAP), a token authenticated
       using the AKC and applied to the CPU configuration to activate a new
       feature.
    
    3. Read the SDSi State Certificate, containing the CPU configuration
       state.
    
    The operations perform function specific mailbox commands that forward the
    requests to SDSi hardware to perform authentication of the payloads and
    enable the silicon configuration (to be made available after power
    cycling).
    
    The SDSi device itself is enumerated as an auxiliary device from the
    intel_vsec driver and as such has a build dependency on CONFIG_INTEL_VSEC.
    
    Link: https://github.com/intel/intel-sdsi
    Signed-off-by: David E. Box <david.e.box@linux.intel.com>
    Reviewed-by: Mark Gross <markgross@kernel.org>
    Link: https://lore.kernel.org/r/20220212013252.1293396-2-david.e.box@linux.intel.com
    Reviewed-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Hans de Goede <hdegoede@redhat.com>

diff --git a/drivers/platform/x86/intel/Makefile b/drivers/platform/x86/intel/Makefile
index 053edc457ccb..1a5cdba4c2d2 100644
--- a/drivers/platform/x86/intel/Makefile
+++ b/drivers/platform/x86/intel/Makefile
@@ -25,6 +25,8 @@ intel_int0002_vgpio-y			:= int0002_vgpio.o
 obj-$(CONFIG_INTEL_INT0002_VGPIO)	+= intel_int0002_vgpio.o
 intel_oaktrail-y			:= oaktrail.o
 obj-$(CONFIG_INTEL_OAKTRAIL)		+= intel_oaktrail.o
+intel_sdsi-y				:= sdsi.o
+obj-$(CONFIG_INTEL_SDSI)		+= intel_sdsi.o
 intel_vsec-y				:= vsec.o
 obj-$(CONFIG_INTEL_VSEC)		+= intel_vsec.o